actors/builtin/miner/miner_state.go

```diff
@@ -267,64 +294,439 @@ func (st *State) ForEachSector(store adt.Store, f func(*SectorOnChainInfo)) erro
 	})
 }
 
-func (st *State) GetStorageWeightDescForSector(store adt.Store, sectorNo abi.SectorNumber) (*power.SectorStorageWeightDesc, error) {
-	sectorInfo, found, err := st.GetSector(store, sectorNo)
+// Adds some sector numbers to the new sectors bitfield.
+func (st *State) AddNewSectors(sectorNos ...abi.SectorNumber) (err error) {
+	ns := abi.NewBitField()
+	for _, sector := range sectorNos {
+		ns.Set(uint64(sector))
+	}
+	st.NewSectors, err = bitfield.MergeBitFields(st.NewSectors, ns)
 	if err != nil {
-		return nil, err
-	} else if !found {
-		return nil, errors.Errorf("no such sector %v", sectorNo)
+		return err
 	}
 
-	return asStorageWeightDesc(st.Info.SectorSize, sectorInfo), nil
+	count, err := st.NewSectors.Count()
+	if err != nil {
+		return err
+	}
+	if count > NewSectorsPerPeriodMax {
+		return fmt.Errorf("too many new sectors %d, max %d", count, NewSectorsPerPeriodMax)
+	}
```

Comment:
`st.NewSectors`'s value is modified by this method (line 303) even if it errors, is that the intended effect?

Reply:
It was, but you're right to feel uneasy about it. https://github.com/filecoin-project/specs-actors/issues/300